log.go

```diff
@@ -2,12 +2,119 @@ package didplc
 
 import (
 	"fmt"
+	"sync"
 	"time"
 
-	"github.com/bluesky-social/indigo/atproto/crypto"
 	"github.com/bluesky-social/indigo/atproto/syntax"
 )
 
+type opStatus struct {
+	CreatedAt   time.Time // the only immutable field here
+	Nullified   bool
+	LastChild   string // CID
+	AllowedKeys []string
+}
+
+// Note: logValidationContext is designed such that it could later be turned into an interface,
+// optionally backed by a db rather than in-memory
+// Note: ops are globally unique by CID, so opStatus map can be shared across all DIDs
```

Comment:
Argh! I'm not sure this is correct. While ops *are* globally unique by CID, I do need to track which ops belong to which DID - otherwise you could update an op using a `prev` from a different DID!

I'll fix this by adding a `did` field to the `opStatus` struct so that it can be checked.